app/services/et_api_handler.rb

```diff
@@ -6,6 +6,7 @@ def self.submit(form_hash)
 
     data_array = [build_response_data(form_hash), build_respondent_data(form_hash)]
     data_array.push(build_representative_data(form_hash)) if form_hash[:your_representative_answers][:have_representative]
+    data_array[1][:data].merge!(build_disability_data(form_hash)) if form_hash[:disability_answers][:disability]
```

Comment:
This is a bit of a code smell

You have to go back and determine what data_array[1] is - which is build_respondent_data(form_hash)

Then, you are doing a merge! on it which is changing the original potentially depending on if build_respondent_data produces its own hash or returns part of form_hash.

Would be much better to do (please choose better naming of vars than me lol)

```
a = build_respondent_data(form_hash)
a = a.merge(build_disability_data(form_hash)) if form_hash[:disability_answers][:disability]
data_array = [build_response_data(form_hash), a]

```
This not only reads much better, but it produces a copy of the hash with merged in data rather than mutating the original.  Its up to you if you do that or merge! it - im just highlighting the potential issue.

Reply:
Thanks for picking this up, fixed in f264597f272944608a624dbc2648d28607123554